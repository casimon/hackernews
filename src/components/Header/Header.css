.Header {
  background-color: #282c34;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: flex-start;
  font-size: calc(10px + 2vmin);
  color: white;
}

.Header h1 {
  margin: 0;
  padding: 0;
  font-size: 32px;
}

.Header img {
  width: 100px;
  height: 100px;
  margin-right: 20px;
}

.Header-nav {
  padding: 20px;
}

.Header-nav a {
  color: #fb651e;
  font-size: 18px;
  text-decoration: none;
  padding: 10px 20px 10px 20px;
}
