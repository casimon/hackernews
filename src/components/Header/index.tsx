import React from 'react'
import './Header.css'
import Logo from '../../assets/Y_Combinator_logo.svg'

const Header = () => {
  return (
    <header className='Header'>
      <img src={Logo} alt='Logo' />
      <h1>Hacker News</h1>
      <div className='Header-nav'>
        <a href='/'>Home</a>
        <a href='/comments'>Comments</a>
      </div>
    </header>
  )
}

export default Header
