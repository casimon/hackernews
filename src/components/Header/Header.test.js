import React from 'react'

import { render } from '@testing-library/react'
import Header from './Header'

const init = (props) => render(<Header {...props} />)

describe('Test header component', () => {
  test('should render header with no props', () => {
    const { container } = init()
    expect(container).toMatchSnapshot()
  })
})
