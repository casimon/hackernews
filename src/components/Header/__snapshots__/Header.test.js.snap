// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test header component should render header with no props 1`] = `
<div>
  <header
    class="Header"
  >
    <img
      alt="Logo"
      src="Y_Combinator_logo.svg"
    />
    <h1>
      Hacker News
    </h1>
    <div
      class="Header-nav"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/comments"
      >
        Comments
      </a>
    </div>
  </header>
</div>
`;
